web/thesauruses/javascript/ECMAScript 2009/classes.json

```diff
@@ -0,0 +1,108 @@
+      "name": "Defining public variables"
+    },
+    "static_variables": {
+      "code": "var ClassName = (function() {\n function ClassName() {\n  // Constructor body\n }\n ClassName.staticField = 0;\n return ClassName;\n})();",
+      "name": "Defining static variables"
+    },
+    "private_functions": {
+      "code": "var ClassName = (function() {\n var privateMethod = function() {\n  // Method body\n }\n // privateMethod is accessible in this scope only.\n function ClassName() {\n  // Constructor body\n }\n return ClassName;\n})();",
+      "name": "Defining private functions"
+    },
+    "protected_functions": {
+      "not-implemented": true,
+      "name": "Defining protected functions"
+    },
+    "public_functions": {
+      "code": "var ClassName = (function() {\n function ClassName() {\n  // Constructor body\n }\n ClassName.prototype.publicMethod = function() {\n  // Method body\n }\n return ClassName;\n})();",
+      "name": "Defining public functions"
+    },
+    "static_functions": {
+      "code": "var ClassName = (function() {\n function ClassName() {\n  // Constructor body\n }\n ClassName.staticMethod = function() {\n  // Method body\n }\n return ClassName;\n})();",
+      "name": "Defining static functions"
+    },
+    "extends_class": {
+      "code": "var ParentClassName = function ParentClassName() {\n // Constructor body\n}\nvar ChildClassName = (function() {\n function ChildClassName() {\n  // Call the parent class constructor, should be first line of constructor\n  ParentClassName.call(this);\n  // Constructor body\n }\n // Copy the prototype from the parent class\n ChildClassName.prototype = Object.create(ParentClassName.prototype);\n // Set the constructor to refer to this class\n ChildClassName.prototype.constructor = ChildClassName;\n return ChildClassName;\n})();",
+      "name": "Class that inherits/extends another class"
+    },
+    "extending_interface": {
+      "not-implemented": true,
+      "name": "Class/Interface that inherits/extends another class/interface"
+    },
+    "calling_superclass_functions": {
+      "code": "var ParentClassName = function ParentClassName() {\n // Parent class constructor\n};\nParentClassName.prototype.publicMethod = function() {\n // Method body\n};\nvar ChildClassName = (function() {\n function ChildClassName() {\n  // Call the parent class constructor, should be first line of constructor\n  ParentClassName.call(this);\n }\n // Copy the prototype from the parent class\n ChildClassName.prototype = Object.create(ParentClassName.prototype);\n // Set the constructor to refer to this class\n ChildClassName.prototype.constructor = ChildClassName;\n // Override the publicMethod function\n ChildClassName.prototype.publicMethod = function() {\n  // Call the superclass function, which is publicMethod of ParentClassName prototype.\n  ParentClassName.prototype.publicMethod.call(this);\n};\n return ChildClassName;\n})();",
```

Comment:
This really could just be as simple as `ParentClassName.prototype.publicMethod.call(this);`, right? The hope is to create _just_ the stuff needed to perform these actions. It's hopefully obvious ParentClassName would be a parent class, for instance.

Reply:
Updated in [7b85db9](https://github.com/codethesaurus/codethesaur.us/pull/569/commits/7b85db9831e0297caec59959931bc69fc278cc2e)